docs/non-normative/how-to-write-conventions/resource-and-entities.md

```diff
@@ -0,0 +1,301 @@
+
+### Declaring associations between signals
+
+You can declare which entities should be used with specific observability
+signals.  For example, process metrics should be used with the process
+entity, so that the metric is associated with a known process. To declare
+this, use the `entity_associations` field on the signal and reference
+another resource group *by name*.
+
+`model/{my_domain}/metrics.yaml`:
+
+```yaml
+groups:
+  - id: metric.some_metric
+    type: metric
+    ...
+    entity_associations: 
+      - my_entity
+```
+
+Notes:
+
+- You cannot declare an association on an *unstable* resource from a
+  *stable* signal.
+- You can declare multiple associations. These form a "one or many" set,
+  where one or many of the named entities may be associated with the
+  metric. There is *no* requirement on disjointness due to how is-a
+  relationships are modelled.
+
+### Extending an entity
+
+While not recommended for Semantic Conventions, you can define a new
+"view" of an entity that includes additional descriptive attributes. To
+do so, use the extends field on groups:
+
+`model/{my_other_domain}/entities.yaml`:
+
+```yaml
+groups:
+  - id: entity.my_entity_2
+    type: entity
+    extends: entity.my_entity
+    attributes:
+      - ref: new.attribute.name
+        requirement_level: opt_in
+        role: descriptive
+```
+
+Notes:
+
+- You cannot change the `name` or `type` fields of the new entity.
+- You cannot change the set of identifying attributes.
+
+## FAQ
+
+### When to define a new entity?
+
+There are two scenarios where entities should be defined:
+
+- When you are generating a new signal (log, metric, span, etc.)
+  and no existing entity makes sense as the "source".
+- (future) When you need to describe an entity hierarchy from
+  some system of record (e.g. resources in kubernetes, assets
+  in a cloud).
+
+For example, the Mainframe SIG in OpenTelemetry will need to outline a
+set of entities and relationships to define observed telemetry. Another example:
+If a new clustering solution (e.g. Hashicorp's Nomad) is defined, and existing
+container-based entities are not enough, then new entities should be
+defined.
+
+### When to define an "is-a" relationship vs. extending descriptive attributes?
+
+There are two key rules:
+
+- Default to introducing separate entities with a clear "is a" (or similar) relationship
+- Extend an entity with new descriptive attributes if and only if the following is true:
+  - The extending entity cannot be associated with any telemetry by itself.
+    *For example, when adding a `windows.process` entity, but no new
+    signals (metrics, logs, spans, etc.) would be added to the entity.*
+  - The extending entity doesn’t have any other entities that logically can
+    only be associated with it.
+    *Note: this only applies to entities as signals with relationships.*
+
+This avoids the complexities of subtyping and ambiguous attribute usage.
```

Comment:
Are we calling "extending" a "subtyping"? Are these alias terms or different terms? If aliases can we choose one and stick with one?